tutorials/plot_dics.py

```diff
@@ -0,0 +1,272 @@
+# Now we run the signal through the forward model to obtain simulated sensor
+# data. To save computation time, we'll only simulate gradiometer data. You can
+# try simulating other types of sensors as well.
+
+# Load the forward model
+fwd = mne.read_forward_solution(fwd_fname)
+
+# Use only gradiometers
+fwd = mne.pick_types_forward(fwd, meg='grad', eeg=False, exclude='bads')
+
+# Create an info object that holds information about the sensors (their
+# location, etc.).
+info = mne.create_info(fwd['info']['ch_names'], sfreq, ch_types='grad')
+info.update(fwd['info'])
+
+# To simulate the data, we need a version of the forward solution where each
+# source has a "fixed" orientation, i.e. pointing orthogonally to the surface
+# of the cortex.
+fwd_fixed = mne.convert_forward_solution(fwd, force_fixed=True)
+
+# Now we can run our simulated signal through the forward model, obtaining
+# simulated sensor data.
+sensor_data = mne.apply_forward_raw(fwd_fixed, stc, info).get_data()
+
+# We're going to add some noise to the sensor data
+noise = np.random.randn(*sensor_data.shape)
+
+# Scale the noise to be in the ballpark of MEG data
+noise_scaling = np.linalg.norm(sensor_data) / np.linalg.norm(noise)
+noise *= noise_scaling
+
+# Mix noise and signal with the given signal-to-noise ratio.
+sensor_data = SNR * sensor_data + noise
+
+###############################################################################
+# We create an :class:`mne.EpochsArray` object containing two trials: one with
+# just noise and one with both noise and signal. The techniques we'll be
+# using in this tutorial depend on being able to contrast data that contains
+# the signal of interest versus data that does not.
+epochs = mne.EpochsArray(
+    data=np.concatenate(
+        (noise[np.newaxis, :, :],
+         sensor_data[np.newaxis, :, :]),
+        axis=0),
+    info=info,
+    events=np.array([[0, 0, 1], [10, 0, 2]]),
+    event_id=dict(noise=1, signal=2),
+)
+
+# Plot the simulated data
+epochs.plot()
+
+###############################################################################
+# Power mapping
+# -------------
+# With our simulated dataset ready, we can now pretend to be researchers that
+# have just recorded this from a real subject and are going to study what parts
+# of the brain communicate with each other.
+#
+# First, we'll create a source estimate of the MEG data. We'll use both a
+# straightforward MNE-dSPM inverse solution for this, and the DICS beamformer
+# which is specifically designed to work with oscillatory data.
+
+###############################################################################
+# Computing the inverse using MNE-dSPM:
+
+# Estimating the noise covariance on the trail that only contains noise.
+cov = mne.compute_covariance(epochs['noise'])
+inv = make_inverse_operator(epochs.info, fwd, cov)
+
+# Apply the inverse model to the trial that also contains the signal.
+s = apply_inverse(epochs['signal'].average(), inv)
```

Comment:
naive question, why don't you use `compute_source_psd_epochs` with `fmin` and `fmax` around 10 Hz?

Reply:
Good idea!

Reply:
Unfortunately, using `compute_source_psd_epochs` caused the memory footprint of the example to skyrocket:
![mem1](https://user-images.githubusercontent.com/428273/34661412-3147b6f0-f452-11e7-8add-d2494d1a33a7.png)

It reached CircleCL's 4GB memory limit. I've reverted back to doing a RMS of the signal, which reduces memory usage:

![mem2](https://user-images.githubusercontent.com/428273/34661423-4803dc8e-f452-11e7-969a-8cb8287a4a92.png)